lib/matplotlib/ticker.py

```diff
@@ -1214,12 +1206,43 @@ class EngFormatter(Formatter):
          24: "Y"
     }
 
-    def __init__(self, unit="", places=None):
+    def __init__(self, unit="", places=None, sep=" "):
+        """
+        Parameters
+        ----------
+        unit : str (default: "")
+            Unit symbol to use, suitable for use with single-letter
+            representations of powers of 1000. For example, 'Hz' or 'm'.
+
+        places : int (default: None)
+            Precision with which to display the number, specified in
+            digits after the decimal point (there will be between one
+            and three digits before the decimal point). If it is None,
+            the formatting falls back to the floating point format '%g',
+            which displays up to 6 *significant* digits (i.e. the
+            equivalent value for *places* varies between 0 and 5, both
```

Comment:
If we want to go this way, I´d rather just replace the `places` argument with something like `format`, which would defaults to `.6g` (or `g` if I understand correctly). But as you warned, it would definitively be a break of the current API ;).